querier.go

```diff
@@ -266,33 +266,88 @@ func (q *blockQuerier) Close() error {
 // checked to not exist in series the postings list points to.
 // It returns EmptyPostings() if it can be determined beforehand that no results will be found.
 func PostingsForMatchers(ix IndexReader, ms ...labels.Matcher) (index.Postings, error) {
-	var its []index.Postings
+	var its, notIts []index.Postings
+	// See which labelnames must be non-empty.
+	labelnameMustBeSet := make(map[string]bool, len(ms))
+	for _, m := range ms {
+		if !m.Matches("") {
+			labelnameMustBeSet[m.Name()] = true
```

Comment:
`m.Matches("")` checks the label value and not the label name so the variable should be named `labelValueMustBeSet`


and don't understand why do you have to check it twice in `for _, m := range ms {`

Reply:
Changed. They're different tests.